### PR TITLE
Add API documentation

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes sul-dlss/dlme#246

### CODE REVIEW

File: lib/macros/dlme.rb [hunk 1]
```diff
@@ -29,6 +39,14 @@ def identifier_with_prefix(context, identifier)
       end
     end
 
+    # Create an identifier that can be used in case none is encoded in the record.
+    # It will first try to take it from the ++command_line.filename++ setting and if
+    # That is not available from the `identifier` setting.
```
Comment (justinlittman) [inline lib/macros/dlme.rb]: lowercase t.

File: lib/macros/met_csv.rb [hunk 2]
```diff
@@ -5,21 +5,34 @@
 module Macros
   # Macros for extracting values from CSV rows
   module MetCsv
-    # @param header_or_index [String] the field header or index to accumulate
-
     DISPLAY_NAME = 'Artist Display Name'.freeze
```
Comment (justinlittman) [inline lib/macros/met_csv.rb]: Isn't freeze redundant? (Yes, I know you didn't touch this code.)

File: lib/macros/dlme.rb (hunk 1 above)
Comment (jcoyne) [reply lib/macros/dlme.rb]: Huh?

Comment (jcoyne) [reply lib/macros/dlme.rb]: Oh, I see.

**Pull request merged**